Update AutoFlatpaks to v1.1.0

jurassicplayer opened the pull request:
# AutoFlatpaks

Please include a summary of what the plugin does.

Major changes under the hood everywhere.
- Now using EventTarget and EventHandlers for passing along information in a sane (hopefully) way that can trigger UI updates
- Added initial flatpak manager page
  - Includes filtering on search term (name, description, reference, origin), package type (app, runtime), status (queued, installed, not installed, updateable), masked/unmasked
  - Includes simple sorting for A to Z, Z to A, Large to Small, and Small to Large
- Auto updates do NOT update masked packages (that's what masks are for)
- Utilizes a queue in the background to handle updates and package changes so you can do other things while it's working
- Live progress is now shown in the QAM menu on package changes for clarity of what is happening
- Shouldn't have issues when offline (at least the package manager shouldn't)

## Checklist:

- [x] I am the original author of this plugin.
- [x] I made my code efficient and readable.
- [x] I have verified that my plugin works properly on the latest versions of SteamOS and decky-loader.

OuterWinnie commented:
Working fine, in preview.

nabel0 reviewed (changes requested):
Got the "blur" problem on the latest SteamOS preview...

jurassicplayer commented:
@nabel0 AutoFlatpaks v1.2.0 should fix the modal issue (added a fallback modal for when Valve breaks DFL's modal, since it's happened twice now).

Note to testers: There is a small change to notifications to use the built-in Toaster available in decky-loader v2.4.6-pre3+ and the original way for v2.4.5 & below, but will break in 2.4.6-pre2/1. Please make sure to update to the latest decky pre-release if testing for pre-release.

rawdatafeel commented:
Decky version: Prerelease, testing, V2.4.6-Pre4

SteamOS Version: Beta, 3.5, 5-13.0-Valve35-1-Neptune

1. Any major blocking issues that you encountered, if any: 
- None, been using it for about a day now, everything works as expected. 
- The new shop menu button works great, log is easy to read and is up to date, buttons in the shop all work as far as I can tell.
      -  "shop" may not be the right word here.
- "Toast" and "Sound" notifications have been on for the past day, and these both seem to work as intended from what I can tell. 
- Updating Flatpaks on the plugin menu gives the names of the flatpaks being updated which is handy and seems to work as intended. 

2. Any minor issues that you encountered (usually cosmetic, don't harm functionality).
- The icons in the shop menu are a little confusing. I was a little confused on what the "eyes" and queuing refers to. I believe trashing = uninstall? That did work and removed a Flatpak from the list. Maybe a "How to Use" tab in this menu would be useful going over the different functionalities. 

3. Yay/nay on to send to the production store or if it requires more testing
 - Yay

jurassicplayer commented:
>     2. Any minor issues that you encountered (usually cosmetic, don't harm functionality).
> 
> 
>     * The icons in the shop menu are a little confusing. I was a little confused on what the "eyes" and queuing refers to. I believe trashing = uninstall? That did work and removed a Flatpak from the list. Maybe a "How to Use" tab in this menu would be useful going over the different functionalities.

Not a bad idea. I've added a basic how-to for using the browse page (I don't really know what I want to call it either) that goes a little bit over the filters, masks, and what the buttons do. Hopefully it's something others can understand, but it'll be there in the next release.
![thumbnail08](https://user-images.githubusercontent.com/5753435/208304408-cc96788b-530b-4b94-a816-0c7849c55240.png)
As for what the eyes are, it's flatpak masking. Put simply, it's to hide the masked packages from the auto-updating for people who want to keep certain packages at a specific version without being bothered about it.

nabel0 reviewed (changes requested):
The "Browse" Screen just shows a black screen form me.
Tested on steamos preview

TrainDoctor commented:
Testing on hold until bugs resolved.

nabel0 reviewed (approved):
Tested on steamos stable and preview. Everything is working fine now.

nabel0 reviewed (approved):
Works fine

nabel0 reviewed (approved):
Works fine on steamos stable and preview

nabel0 commented:
> I've added a basic how-to for using the browse page

Cant see this "page" in the latest version here on the  testing store.
Looks like this for me:

<img width="640" alt="SteamLink_0b4td6wHCE" src="https://user-images.githubusercontent.com/50377455/208483027-cc10c157-1c61-45b9-b757-7d20c8dfca87.png">

nabel0 reviewed (approved):
Tested and working fine on steamos stable and preview
